### PR TITLE
add binary

### PR DESCRIPTION
**Pull request by Oleksii-Lasiichuk**

### CODE REVIEW

File: 3/gray_code.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+bi_number = input()
```
Comment (ivan-pifagor) [inline 3/gray_code.py:1]: Adding task description on top of the file, would help reviewing process:

```
"""
Problem: 

input binary number find ....
"""
```

File: 3/gray_code.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+number = int(bi_number, 2)
+
+len = len(bi_number)
+guessing = number>>(len-1)
+if (guessing)&1==1:
```
Comment (ivan-pifagor) [inline 3/gray_code.py:6]: Why??

`&1==1`

```suggestion
if (guessing):
```

File: 3/gray_code.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    gray = "0"
+for i in range(len-2, -1, -1):
+    count_1 = (number >> i+1) & 1
+    count_2 = (number >> i) & 1
+    compare = count_1^count_2
+    if compare==1:
+        gray += "1"
+    if compare==0:
+        gray += "0"
+print(gray)
```
Comment (ivan-pifagor) [inline 3/gray_code.py:18]: What does "gray" stand for?

File: 3/gray_code.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+guessing = number>>(len-1)
+if (guessing)&1==1:
+    gray = "1"
+else:
+    gray = "0"
+for i in range(len-2, -1, -1):
+    count_1 = (number >> i+1) & 1
+    count_2 = (number >> i) & 1
+    compare = count_1^count_2
+    if compare==1:
```
Comment (ivan-pifagor) [inline 3/gray_code.py:14]:
```suggestion
    if compare == 1:
```

Add whitespace please

File: 3/series.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
+len = int(input())
+series = ""
+operation = True
+for i in range (1, len*2+1, 2):
+    series += f"{i}/{i+1}"
+    if i+1 == len*2:
+        pass
+    else:
+        if operation==True:
```
Comment (ivan-pifagor) [inline 3/series.py:9]: Really? 

```suggestion
        if operation:
```
It doesn't make any sense to compare a boolean for True!!!

**Review by ivan-pifagor — CHANGES_REQUESTED** [review]

File: 3/gray_code.py (hunk 3 above)
Comment (Oleksii-Lasiichuk) [reply 3/gray_code.py:18]: Gray code.
link for wiki:
https://en.wikipedia.org/wiki/Gray_code

File: 3/gray_code.py (hunk 2 above)
Comment (Oleksii-Lasiichuk) [reply 3/gray_code.py:6]: Gray numbers always start with the same digit as the binary number they are compared to. So i am checking whether the first digit of number is 1 or 0

**Pull request merged**